Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -145,8 +145,217 @@ public struct HTTPResponseHead: Equatable {
     }
 }
 
-fileprivate typealias HTTPHeadersStorage = [String: [(String, String)]] // [lowerCasedName: [(originalCaseName, value)]
+/// The Index for a header name or value that points into the underlying `ByteBuffer`.
+typealias HTTPHeaderIndex = (start: Int, length: Int)
```

Comment:
@weissi @Lukasa we could also use a struct here and store the parsed String into it once we parsed it one time. WDYT ?